tiatoolbox/cli/semantic_segment.py

```diff
@@ -0,0 +1,145 @@
+"""Command line interface for semantic segmentation."""
+import click
+import yaml
+
+from tiatoolbox import utils
+from tiatoolbox.models.controller.semantic_segmentor import (
+    IOSegmentorConfig,
+    SemanticSegmentor,
+)
+
+
+@click.group()
+def main():  # pragma: no cover
+    """Define semantic_segment click group."""
+    return 0
+
+
+@main.command()
+@click.option(
+    "--pretrained_model",
+    help="Predefined model used to process the data. the format is "
+    "<model_name>_<dataset_trained_on>. For example, `resnet18-kather100K` is a "
+    "resnet18 model trained on the kather dataset. For a detailed list of "
+    "available pretrained models please see "
+    "https://tia-toolbox.readthedocs.io/en/latest/usage.html"
+    "#tiatoolbox.models.classification.patch_predictor.get_pretrained_model",
+    default="fcn-tissue_mask",
+)
+@click.option(
+    "--pretrained_weights",
+    help="Path to the model weight file. If not supplied, the default "
+    "pretrained weight will be used.",
+    default=None,
+)
+@click.option(
+    "--img_input",
+    help="Path to the input directory containing images to process or an "
+    "individual file.",
+)
+@click.option(
+    "--file_types",
+    help="File types to capture from directory. "
+    "default='*.png', '*.jpg', '*.jpeg', '*.tif', '*.tiff'",
+    default="*.png, *.jpg, *.jpeg, *.tif, *.tiff, *.svs, *.ndpi, *.jp2, *.mrxs",
+)
+@click.option(
+    "--masks",
+    help="Path to the input directory containing masks to process corresponding to "
+    "image tiles and whole-slide images. Patches are only processed if they are "
+    "within a masked area. If masks are not provided, then a tissue mask will be "
+    "automatically generated for whole-slide images or the entire image is "
+    "processed for image tiles. Supported file types are jpg, png.",
+    default=None,
+)
+@click.option(
+    "--mode",
+    help="Type of input to process. Choose from either patch, tile or wsi. Default=wsi",
+    default="wsi",
+)
+@click.option(
+    "--output_path",
+    help="Output directory where model predictions will be saved.",
+    default="semantic_segmentation",
+)
+@click.option(
+    "--batch_size",
+    help="Number of images to feed into the model each time.",
+    default=1,
+)
+@click.option(
+    "--yaml_config_path",
```

Comment:
If we are following POSIX conventions, then we should probably use dash separated words instead of underscores for options (key word arguments) on the command line. See the background sections of python optparse and click documentation etc. for information:
- https://docs.python.org/3/library/optparse.html#background
- https://click.palletsprojects.com/en/7.x/options/
- https://stackoverflow.com/questions/1253679/should-command-line-options-in-posix-style-operating-systems-be-underscore-style
- https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

Reply:
Will be addressed in https://github.com/TIA-Lab/tiatoolbox/pull/161